packages/javascript/bh-shared-ui/src/views/SSOConfiguration/SSOConfiguration.tsx

```diff
@@ -152,27 +170,48 @@ const SSOConfiguration: FC = () => {
         }
     };
 
-    const createSAMLProvider = async (samlProvider: CreateSAMLProviderFormInputs) => {
-        setCreateProviderError('');
+    const upsertSAMLProvider = async (samlProvider: UpsertSAMLProviderFormInputs) => {
+        setUpsertProviderError('');
         try {
-            await apiClient.createSAMLProviderFromFile({ ...samlProvider, metadata: samlProvider.metadata[0] });
+            const payload = { name: samlProvider.name, metadata: samlProvider.metadata && samlProvider.metadata[0] };
+            if (ssoProviderIdToDeleteOrUpdate) {
+                await apiClient.updateSAMLProviderFromFile(ssoProviderIdToDeleteOrUpdate, payload);
+            } else {
+                if (payload.name && payload.metadata) {
+                    await apiClient.createSAMLProviderFromFile({ name: payload.name, metadata: payload.metadata });
+                }
+            }
             listSSOProvidersQuery.refetch();
             closeDialog();
         } catch (error) {
             console.error(error);
-            setCreateProviderError('Unable to create new SAML Provider configuration. Please try again.');
+            setUpsertProviderError(
+                `Unable to ${ssoProviderIdToDeleteOrUpdate ? 'update' : 'create new'} SAML Provider configuration. Please try again.`
+            );
         }
     };
 
-    const createOIDCProvider = async (oidcProvider: CreateOIDCProviderRequest) => {
-        setCreateProviderError('');
+    const upsertOIDCProvider = async (oidcProvider: UpsertOIDCProviderRequest) => {
+        setUpsertProviderError('');
         try {
-            await apiClient.createOIDCProvider(oidcProvider);
+            if (ssoProviderIdToDeleteOrUpdate) {
+                await apiClient.updateOIDCProvider(ssoProviderIdToDeleteOrUpdate, oidcProvider);
+            } else {
+                if (oidcProvider.name && oidcProvider.client_id && oidcProvider.issuer) {
```

Comment:
do we need this check in the case we are updating as well?

Reply:
Nah the API doesn't need fields since it's a PATCH endpoint it accepts partial payloads